wallet-services/src/main/java/org/exoplatform/wallet/service/WalletAccountServiceImpl.java

```diff
@@ -500,6 +500,43 @@ public void switchWalletProvider(long identityId,
     }
   }
 
+  @Override
+  public Wallet createWalletInstance(WalletProvider provider, String address, long identityId) throws IllegalAccessException {
+    if (provider == null) {
+      throw new IllegalArgumentException("provider is mandatory");
+    }
+    if (StringUtils.isBlank(address)) {
+      throw new IllegalArgumentException("Wallet address is empty, thus it can't be saved");
+    }
+
+    Identity identity = getIdentityById(identityId);
+    WalletType type = WalletType.getType(identity.getProviderId());
+    if (type.isSpace() && provider != WalletProvider.INTERNAL_WALLET) {
+      throw new IllegalStateException("Can't create wallet with provider " + provider.name() + "for a space"
+          + identity.getRemoteId());
+    }
+    Wallet wallet = new Wallet();
+    wallet.setAddress(address);
+    wallet.setProvider(provider.name());
+    wallet.setTechnicalId(identityId);
+    wallet.setType(type.name());
+    computeWalletIdentity(wallet);
+    Wallet oldWallet = accountStorage.getWalletByIdentityId(wallet.getTechnicalId(), getContractAddress());
+    boolean isNew = oldWallet == null;
+
+    checkCanSaveWallet(wallet, oldWallet, identity.getRemoteId());
+    if (isNew) {
+      wallet.setInitializationState(WalletState.NEW.name());
+    } else if (!StringUtils.equalsIgnoreCase(oldWallet.getAddress(), wallet.getAddress())) {
+      wallet.setInitializationState(WalletState.MODIFIED.name());
+    } else {
+      throw new IllegalAccessException("Can't modify wallet properties once saved");
+    }
+    wallet.setEnabled(isNew || oldWallet.isEnabled());
+    setWalletPassPhrase(wallet, oldWallet);
+    return accountStorage.saveWallet(wallet, isNew);
```

Comment:
this method should be about creating a wallet instance without storing it IMO. In addition, if this is not a bout a new wallet, we should preserve the technical identifier coming from `oldWallet` too.